cfgov/scripts/unpublish_closed_jobs.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import urllib2
+import requests
+import json
+import os
+
+from bs4 import BeautifulSoup
+
+from jobmanager.models.pages import JobListingPage
+
+logger = logging.getLogger(__name__)
+
+URL = os.environ.get('USAJOBS_ARCHIVE_URL')
+HEADERS = {
+    'Host': os.environ.get('USAJOBS_HOST'),
+    'User-Agent': os.environ.get('USAJOBS_USER'),
+    'Authorization-Key': os.environ.get('USAJOBS_API_KEY'),
+}
+
+
+def job_page_closed(link):
+    webpage = urllib2.urlopen(link)
+    soup = BeautifulSoup(webpage, 'html.parser')
+    closed_text = soup.find('div', attrs={'class': 'usajobs-joa-closed'})
+    if closed_text:
+        return 'This job announcement has closed' in closed_text.contents
+
+
+def job_archived(link):
+    try:
+        control_number = os.path.basename(os.path.normpath(link))
+        response = requests.get(URL, headers=HEADERS, params={
+            'ControlNumber': control_number,
+            'WhoMayApply': 'all'
+        })
+        results = json.loads(response.text)['SearchResult']
+        search_items = results['SearchResultItems']
+        return search_items[0]['MatchedObjectId'] == control_number
+    except:
```

Comment:
It would be nice to specify what exceptions you expect here, and maybe log something on the following line so that there isn't an unexpected failure that can't be diagnosed when the scheduled job runs.

Reply:
I added some specific exception logging in https://github.com/cfpb/cfgov-refresh/pull/3787/commits/8a577f1f4c5d79295c6d8e8f9e9eb2b83b153e94.

Reply:
@virginiacc two minor suggestions:

1. You can use `logger.exception` instead of `logger.info` to automatically include the exception in the stack trace, without needing to add it to the message.
2. [PEP8 recommends](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) avoiding bare `except:` clauses (instead use `except Exception:`), because they make it harder to exit with Control-C:
    
    > A bare except: clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use except `Exception:` (bare except is equivalent to `except BaseException:`)

Reply:
@chosak I updated to use `logger.exception` and a single `except Exception:` clause, since the exception data is automatically logged (unless it's preferable to use individual handlers?)

Thanks for the suggestions, @willbarton & @chosak!